src/components/Invoice/EditInvoice.tsx

```diff
@@ -0,0 +1,19 @@
+import { useParams } from 'react-router-dom';
+import InvoiceForm from './InvoiceForm';
+import { useInvoice } from './use-invoice';
+import type { IInvoiceForm } from '../Models/InvoiceForm.interface';
+
+export default function EditInvoice() {
+  const { id } = useParams();
+  const { data } = useInvoice(id as string);
```

Comment:
I think this is a perfect place to use zod validation as well (https://github.com/colinhacks/zod). You can parse params and fetched data, so you are sure the data shape is correct.

Reply:
but not error apperas in single invoice

Reply:
This is fine, data should be of type Invoice[] | undefined, as it needs to be fetched before is available :D